lib/Driver/Driver.cpp

```diff
@@ -342,6 +344,13 @@ static bool populateOutOfDateMap(InputInfoMap &map, StringRef argsHashStr,
   }
 
   if (!optionsMatch) {
+    if (ShowIncrementalBuildDecisions) {
+      llvm::outs() << "Incremental compilation has been disabled, because "
+                   << "different arguments were passed to the compiler.\n"
+                   << "\tCompiling with argument hash: " << argsHashStr << "\n"
```

Comment:
I agree. I also think printing "because different arguments were used" would be frustrating -- I'd ideally like to know how they differed. Should I try to print out the delta here, or is that too much work?